services/brig/src/Brig/Data/Client.hs

```diff
@@ -202,6 +210,9 @@ updateClientLabelQuery = "UPDATE clients SET label = ? WHERE user = ? AND client
 selectClientIds :: PrepQuery R (Identity UserId) (Identity ClientId)
 selectClientIds = "SELECT client from clients where user = ?"
 
+selectClientsBulk :: PrepQuery R (Identity [UserId]) (UserId, ClientId, ClientType, UTCTimeMillis, Maybe Text, Maybe ClientClass, Maybe CookieLabel, Maybe Latitude, Maybe Longitude, Maybe Text)
+selectClientsBulk = "SELECT user, client, type, tstamp, label, class, cookie, lat, lon, model from clients where user in ?"
```

Comment:
Using the IN keyword here with 1000 userIds is bad for performance - the targeted cassandra node may need to contact all other nodes in the cluster to return a result.

The userId is the partition key for this table (see https://github.com/wireapp/wire-server/blob/develop/docs/reference/cassandra-schema.cql#L1112-L1124)

For more information see

See https://docs.datastax.com/en/cql-oss/3.3/cql/cql_reference/cqlSelect.html#cqlSelect__selectInNot
and 
https://lostechies.com/ryansvihla/2014/09/22/cassandra-query-patterns-not-using-the-in-query-for-multiple-partitions/

The solution to this performance problem is to mapconcurrently this query. As you probably don't want to spawn 1000 threads, you can use this very handy function from unliftIO called `pooledForConcurrentlyN_` to limit the amount of threads to say 50 or 100.

Reply:
Aha, thanks! For my education, how did you come up with 50 or 100 threads?

Reply:
Accouding to [Brig.App](https://github.com/wireapp/wire-server/blob/a7226b610983cc0a7dccfdee59230fca59358b48/services/brig/src/Brig/App.hs#L395-L404), we have
- 4 stripes
- 4 connections per stripe
- 128 streams per connection ([default](https://hackage.haskell.org/package/cql-io-1.1.1/docs/Database-CQL-IO.html#v:defSettings))

So, theoretically, we can have 2048 (4 * 4 * 128) requests going in parallel. If we say each request to get list of clients spawns 50 cassandra queries  in parallel at max, we're way below that limit. 
According to [the datastax java driver](https://docs.datastax.com/en/developer/java-driver/4.10/manual/core/pooling/) getting more streams doesn't improve much on performance as there is only so much a server can serve in parallel. So, this also suggests that our guess of running 50 requests in parallel is fairly conservative. 
As we don't want one type of request congesting the pool, and this request shouldn't get called with many clients very often, I think we should be fine keeping this conservative, if we see this being slow, let's consider bumping the number of requests then. WDYT?